prov/util/src/util_coll.c

```diff
@@ -764,6 +745,43 @@ static struct fi_ops util_coll_fi_ops = {
 	.ops_open = fi_no_ops_open,
 };
 
+static inline int util_coll_mc_alloc(struct util_coll_mc **coll_mc)
```

Comment:
I would pass in the av_set and endpoint into this function, possibly the context as well.  See above comment about changing this from alloc() to init().